jobs/go/sdk/README.md

```diff
@@ -0,0 +1,145 @@
+  - '== APP - job-service == Job scheduled:  R2-D2'
+  - '== APP - job-service == Job scheduled:  C-3PO'
+  - '== APP - job-scheduler == Get job response:  {"droid":"C-3PO","Task":"Memory Wipe"}'
+  - '== APP - job-service == Job scheduled:  BB-8'
+  - '== APP - job-scheduler == Get job response:  {"droid":"BB-8","Task":"Internal Gyroscope Check"}'
+  - '== APP - job-scheduler == Job deleted:  BB-8'
+  - '== APP - job-service == Starting droid: R2-D2'
+  - '== APP - job-service == Executing maintenance job: Oil Change'
+  - '== APP - job-service == Starting droid: C-3PO'
+  - '== APP - job-service == Executing maintenance job: Memory Wipe'
+expected_stderr_lines:
+output_match_mode: substring
+match_order: none
+background: true
+sleep: 60
+timeout_seconds: 120
+-->
+
+```bash
+dapr run -f .
+```
+
+The terminal console output should look similar to this, where:
+
+- The `R2-D2` job is being scheduled.
+- The `C-3PO` job is being scheduled.
+- The `C-3PO` job is being retrieved.
+- The `BB-8` job is being scheduled.
+- The `BB-8` job is being retrieved.
+- The `BB-8` job is being deleted.
+- The `R2-D2` job is being executed after 5 seconds.
+- The `R2-D2` job is being executed after 10 seconds.
+
+
+```text
+== APP - job-service == dapr client initializing for: 127.0.0.1:6281
+== APP - job-service == Registered job handler for:  R2-D2
+== APP - job-service == Registered job handler for:  C-3PO
+== APP - job-service == Registered job handler for:  BB-8
+== APP - job-service == Starting server on port: 6200
+== APP - job-service == Job scheduled:  R2-D2
+== APP - job-service == Job scheduled:  C-3PO
+== APP - job-service == 2024/07/17 18:09:59 job:{name:"C-3PO"  due_time:"10s"  data:{value:"{\"droid\":\"C-3PO\",\"Task\":\"Memory Wipe\"}"}}
+== APP - job-scheduler == Get job response:  {"droid":"C-3PO","Task":"Memory Wipe"}
+== APP - job-service == Job scheduled:  BB-8
+== APP - job-service == 2024/07/17 18:09:59 job:{name:"BB-8"  due_time:"15s"  data:{value:"{\"droid\":\"BB-8\",\"Task\":\"Internal Gyroscope Check\"}"}}
+== APP - job-scheduler == Get job response:  {"droid":"BB-8","Task":"Internal Gyroscope Check"}
+== APP - job-scheduler == Deleted job:  BB-8
+```
+
+After 5 seconds, the terminal output should present the `R2-D2` job being processed:
+
+```text
+== APP - job-service == Starting droid: R2-D2
+== APP - job-service == Executing maintenance job: Oil Change
+```
+
+After 10 seconds, the terminal output should present the `C3-PO` job being processed:
+
+```text
+== APP - job-service == Starting droid: C-3PO
+== APP - job-service == Executing maintenance job: Memory Wipe
+```
+
+2. Stop and clean up application processes
+
+```bash
+dapr stop -f .
+```
+
+<!-- END_STEP -->
+
+## Run the Jobs APIs individually
+
+### Schedule Jobs
+
+1. Open a terminal and run the `job-service` app:
+
+```bash
+dapr run --app-id job-service --app-port 6200 --dapr-http-port 6280 --dapr-grpc-port 6281 --app-protocol grpc -- go run .
```

Comment:
Do you really need these settings ?
--dapr-http-port 6280 --dapr-grpc-port 6281 --app-protocol grpc
And if only grpc, then why provide the HTTP one?

Reply:
Can you have this instead?
dapr run --app-id job-service --app-port 6200 -- go run .

Reply:
the grpc protocol is handling the job callback when it is triggered. what we don't need is the http port, really. I'll remove it.